### PR TITLE
add fallback for fal.

### PR DESCRIPTION
**Pull request by vwxyzh**
@chenkennt @superyyrrzz @qinezh @vicancy @ansyral @hellosnow

### CODE REVIEW

**Comment by superyyrrzz** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Common/FileAbstractLayer/RealFileReader.cs [hunk 1]
```diff
@@ -55,7 +55,7 @@ public IEnumerable<RelativePath> EnumerateFiles()
         {
             var length = InputFolder.Length;
             return from f in Directory.EnumerateFiles(InputFolder, "*.*", SearchOption.AllDirectories)
-                   select (RelativePath)f.Substring(length);
+                   select ((RelativePath)f.Substring(length)).GetPathFromWorkingFolder();
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.Common/FileAbstractLayer/RealFileReader.cs]: >(RelativePath)f.Substring(length) [](start = 27, length = 33)

(RelativePath)(f.Substring(length))?

File: src/Microsoft.DocAsCode.Common/FileAbstractLayer/FallbackFileReader.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+namespace Microsoft.DocAsCode.Common
+{
+    using System.Collections.Generic;
+    using System.Collections.Immutable;
+    using System.Linq;
+
+    internal sealed class FallbackFileReader : IFileReader
+    {
+        public FallbackFileReader(ImmutableArray<IFileReader> readers)
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.Common/FileAbstractLayer/FallbackFileReader.cs]: >ImmutableArray [](start = 34, length = 14)

is `params` more suitable for a fallback?

File: src/Microsoft.DocAsCode.Common/FileAbstractLayer/FallbackFileReader.cs [hunk 3]
```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.DocAsCode.Common
+{
+    using System.Collections.Generic;
+    using System.Collections.Immutable;
+    using System.Linq;
+
+    internal sealed class FallbackFileReader : IFileReader
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.Common/FileAbstractLayer/FallbackFileReader.cs]: How about `FileReaderWithFallback`?

**Comment by ansyral** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Common/FileAbstractLayer/RealFileReader.cs (hunk 1 above)
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Common/FileAbstractLayer/RealFileReader.cs]: same ...

File: src/Microsoft.DocAsCode.Common/FileAbstractLayer/FallbackFileReader.cs (hunk 3 above)
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Common/FileAbstractLayer/FallbackFileReader.cs]: `Fallbackable` is not a word...

**Comment by hellosnow** [comment]
:shipit:

**Comment by qinezh** [comment]
:shipit:

**Pull request merged**